test/module/shared_model/mock_objects_factories/mock_command_factory.cpp

```diff
@@ -300,5 +300,28 @@ namespace shared_model {
             return specific_cmd_mock;
           });
     }
+
+    MockCommandFactory::FactoryResult<MockCompareAndSetAccountDetail>
+    MockCommandFactory::constructCompareAndSetAccountDetail(
+        const shared_model::interface::types::AccountIdType &account_id,
+        const shared_model::interface::types::AccountDetailKeyType &cmd_key,
+        const shared_model::interface::types::AccountDetailValueType &cmd_value,
+        const boost::optional<
+            shared_model::interface::types::AccountDetailValueType>
+            cmd_old_value) const {
+      return createFactoryResult<MockCompareAndSetAccountDetail>(
+          [&account_id, &cmd_key, &cmd_value, &cmd_old_value](
+              FactoryResult<MockCompareAndSetAccountDetail> specific_cmd_mock) {
+            ON_CALL(*specific_cmd_mock, accountId())
```

Comment:
please use `EXPECT_CALL` and `WillRepeatedly` like above to see less warnings

Reply:
Fixed